extensions/amp-lightbox-gallery/1.0/component.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {Lightbox} from './../../amp-lightbox/1.0/lightbox';
+import {LightboxGalleryContext} from './context';
+import {sequentialIdGenerator} from '../../../src/utils/id-generator';
+import {
+  useContext,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from '../../../src/preact';
+
+const generateLightboxItemKey = sequentialIdGenerator();
+
+/**
+ * @param {!LightboxGalleryDef.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function LightboxGallery({children}) {
```

Comment:
I think the naming is slightly off, but maybe it's just me. This is not a lightbox itself, but a way to get it. IMHO we could just call it `LightboxGalleryProvider`.

Reply:
Done.